web/src/actions/actionCreators.ts

```diff
@@ -116,6 +117,7 @@ export interface ITeamsPayload {
 
 export const getTeams = (league: string) => {
   return (dispatch: ThunkDispatch<IAppState, void, ITeamsPayload>) => {
+    // TODO: don't re-get if it's already in there
```

Comment:
Quite simple to do here if you want:
```
  return (dispatch: ThunkDispatch<IAppState, void, ITeamsPayload>, getState: any) => {
    const { leagueTeams } = getState();
    // check league teams in object
```
But possibly better to put condition in components to only retrieve if teams props not populated.

....or maybe even better to get teams on first app load only and remove `getTeams` from all components?

Reply:
On first app load sounds like the right move, everything's going to use that. Same with "leagues". What's the good way to do that? Connect the [`App`](https://github.com/NathanDeMaria/py-glicko/blob/e2f69c955f47128ba6f0f2398fc662d3af673878/web/src/App.tsx#L49)? Add some code [here in the root `index`](https://github.com/NathanDeMaria/py-glicko/blob/ea2c735a9b88011af87a24a3b5063d8a424e3367/web/src/index.tsx#L11)?

Reply:
Yeah that's where I've done it. If you've got the "loading" notifications in the components anyway, should only need putting in there and removing the load from all components